packages/components/nodes/documentloaders/ConfluenceServer/Confluence.ts

```diff
@@ -0,0 +1,118 @@
+import { ICommonObject, INode, INodeData, INodeParams } from '../../../src/Interface'
```

Comment:
instead of separating into a different loader, can we use the same loader, but just separate the connection logic (i.e `confluenceOptions`) depending if user selected PAT or Access Token?

A similar approach [here](https://github.com/FlowiseAI/Flowise/blob/main/packages/components/nodes/memory/RedisBackedChatMemory/RedisBackedChatMemory.ts#L71)

Reply:
Hello, @HenryHengZJ 

I attempted an approach that, unfortunately, didn't work out as expected due to a specific issue. Below is the approach I tried:

```typescript
import { INodeCredential, INodeParams } from '../src/Interface'

class ConfluenceApi implements INodeCredential {
    label: string;
    name: string;
    version: number;
    description: string;
    inputs: INodeParams[]

    constructor() {
        this.label = 'Confluence API';
        this.name = 'confluenceApi';
        this.version = 1.0;
        this.description = 
            'Choose the type of Confluence platform and refer to the official guide to get the Access Token for Cloud or Personal Access Token for Server.';
        this.inputs = [
            {
                label: 'Platform Type',
                name: 'platformType',
                type: 'options',
                options: [
                    {
                        label: 'Cloud',
                        name: 'cloud'
                    },
                    {
                        label: 'Server',
                        name: 'server'
                    },
                ],
                default: 'cloud'
            },
            {
                label: 'Access Token',
                name: 'accessToken',
                type: 'password',
                placeholder: '<CONFLUENCE_ACCESS_TOKEN>',
                show: {
                    'platformType': ['cloud']
                }
            },
            {
                label: 'Username',
                name: 'username',
                type: 'string',
                placeholder: '<CONFLUENCE_USERNAME>',
                show: {
                    'platformType': ['cloud']
                }
            },
            {
                label: 'Personal Access Token',
                name: 'personalAccessToken',
                type: 'password',
                placeholder: '<CONFLUENCE_PERSONAL_ACCESS_TOKEN>',
                show: {
                    'platformType': ['server']
                }
            }
        ]
    }
}

module.exports = { credClass: ConfluenceApi }
```

**Encountered Issue:**
While trying to implement this, I ran into a TypeScript error: 

```
credentials/ConfluenceApi.credential.ts:56:17 - error TS2322: Type '{ label: string; name: string; type: string; placeholder: string; show: { platformType: string[]; }; }' is not assignable to type 'INodeParams'.
  Object literal may only specify known properties, and 'show' does not exist in type 'INodeParams'.
```

This error indicates that the `show` property is not recognized within the `INodeParams` type, which is crucial for our implementation to dynamically display input fields based on the selected platform type.

I can try to make two different credentials for one loader like in the example that you provided.

Reply:
For the document loader, this code was used as an approach to include a platform selection option in the credentials.

```
import { ICommonObject, INode, INodeData, INodeParams } from '../../../src/Interface'
import { TextSplitter } from 'langchain/text_splitter'
import { ConfluencePagesLoader, ConfluencePagesLoaderParams } from 'langchain/document_loaders/web/confluence'
import { getCredentialData, getCredentialParam } from '../../../src'

class Confluence_DocumentLoaders implements INode {
    label: string
    name: string
    version: number
    description: string
    type: string
    icon: string
    category: string
    baseClasses: string[]
    credential: INodeParams
    inputs: INodeParams[]

    constructor() {
        this.label = 'Confluence'
        this.name = 'confluence'
        this.version = 1.0
        this.type = 'Document'
        this.icon = 'confluence.svg'
        this.category = 'Document Loaders'
        this.description = `Load data from a Confluence Document`
        this.baseClasses = [this.type]
        this.credential = {
            label: 'Connect Credential',
            name: 'credential',
            type: 'credential',
            credentialNames: ['confluenceApi']
        }
        this.inputs = [
            {
                label: 'Text Splitter',
                name: 'textSplitter',
                type: 'TextSplitter',
                optional: true
            },
            {
                label: 'Base URL',
                name: 'baseUrl',
                type: 'string',
                placeholder: 'https://example.atlassian.net/wiki'
            },
            {
                label: 'Space Key',
                name: 'spaceKey',
                type: 'string',
                placeholder: '~EXAMPLE362906de5d343d49dcdbae5dEXAMPLE',
                description:
                    'Refer to <a target="_blank" href="https://community.atlassian.com/t5/Confluence-questions/How-to-find-the-key-for-a-space/qaq-p/864760">official guide</a> on how to get Confluence Space Key'
            },
            {
                label: 'Limit',
                name: 'limit',
                type: 'number',
                default: 0,
                optional: true
            },
            {
                label: 'Metadata',
                name: 'metadata',
                type: 'json',
                optional: true,
                additionalParams: true
            }
        ]
    }

    async init(nodeData: INodeData, _: string, options: ICommonObject): Promise<any> {
        const spaceKey = nodeData.inputs?.spaceKey as string
        const baseUrl = nodeData.inputs?.baseUrl as string
        const limit = nodeData.inputs?.limit as number
        const textSplitter = nodeData.inputs?.textSplitter as TextSplitter
        const metadata = nodeData.inputs?.metadata

        const credentialData = await getCredentialData(nodeData.credential ?? '', options)
        const accessToken = getCredentialParam('accessToken', credentialData, nodeData)
        const username = getCredentialParam('username', credentialData, nodeData)
        // Извлекаем Personal Access Token, если он доступен в учетных данных
        const personalAccessToken = getCredentialParam('personalAccessToken', credentialData, nodeData)

        const confluenceOptions: ConfluencePagesLoaderParams = {
            username,
            accessToken,
            baseUrl,
            spaceKey,
            limit,
        };
        
        // If personalAccessToken is available, use it instead of the regular access token
        if (personalAccessToken) {
            confluenceOptions.personalAccessToken = personalAccessToken;
        }

        const loader = new ConfluencePagesLoader(confluenceOptions);

        let docs = [];

        if (textSplitter) {
            docs = await loader.loadAndSplit(textSplitter);
        } else {
            docs = await loader.load();
        }

        if (metadata) {
            const parsedMetadata = typeof metadata === 'object' ? metadata : JSON.parse(metadata);
            let finaldocs = [];
            for (const doc of docs) {
                const newdoc = {
                    ...doc,
                    metadata: {
                        ...doc.metadata,
                        ...parsedMetadata
                    }
                };
                finaldocs.push(newdoc);
            }
            return finaldocs;
        }

        return docs;
    }
}

module.exports = { nodeClass: Confluence_DocumentLoaders };
```